engineering_tools/carma

```diff
@@ -252,6 +252,70 @@ carma-config__status() {
     fi
 }
 
+carma__exec() {
+    local TARGET_IMAGE="usdotfhwastol/carma-platform:latest"
+    local COMMAND="bash"
+
+    # Check if the -i argument was provided
+    # If the image was specified then use that
+    if [ "$1" = "-i" ]; then 
+
+        if [ -z "$2" ]; then
+            echo " When -i is specified you must enter a carma docker image and tag."
+            exit -1
+        fi
+
+        local TARGET_IMAGE="$2"
+
+        shift
+        shift
+
+    # If the image was not specified use what was in the carma config
+    elif [ docker container inspect carma-config > /dev/null 2>&1 ]; then
+
+        local IMAGE_BACKUP=$TARGET_IMAGE
+        local SUPPORTED_ORANIZATIONS="usdotfhwastol\|usdotfhwastoldev"
+        local SUPPORTED_IMAGES="carma-platform:\|carma-messenger:"
+
+        # sed command finds the first instance of an image which is part of the set defined in SUPPORTED_ORGANIZATIONS and SUPPORTED_IMAGES.
+        # then returns the name of the image.  
+        local TARGET_IMAGE=$(docker run --rm --volumes-from carma-config:ro --entrypoint sh busybox:latest -c \
```

Comment:
At some point this might be worth making into it's own mode, just to get the current configured image name for a service in the docker-compose.

Reply:
Yeah I agree. Though it might take a bit more though. This one is looking for the image name rather than service name so the regex would probably need to be reworked.